File: LeetCode75-sols/Array_String/CPP/Product_of_Array_Except_Self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> answer(size(nums), 1);
        int prodval = 1;
        // Create left products
        for (int i = 1; i < size(nums); i++) {
            answer[i] = answer[i-1] * nums[i-1];
        }
        // Create right products and merge
        for (int i = size(nums) - 1; i >= 0; i--) {
            answer[i] *= prodval;
            prodval *= nums[i];
        }
        return answer;
    }
};

File: LeetCode75-sols/Array_String/Python/Product_of_Array_Except_Self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        answer = [1] * len(nums)
        prodval = 1
        # Create left products
        for i in range(1,len(nums)):
            answer[i] = answer[i-1] * nums[i-1]
        # Create right products and merge
        for i in range(len(nums) - 1, -1, -1):
            answer[i] *= prodval
            prodval *= nums[i]
        return answer